Client/Frontend/Reader/ReaderModeCache.swift

```diff
@@ -18,69 +18,77 @@ class ReaderModeCache {
         return ReaderModeCacheSharedInstance
     }
 
-    func put(url: NSURL, _ readabilityResult: ReadabilityResult, error: NSErrorPointer) -> Bool {
-        if let cacheDirectoryPath = cacheDirectoryForURL(url) {
-            if NSFileManager.defaultManager().createDirectoryAtPath(cacheDirectoryPath as String, withIntermediateDirectories: true, attributes: nil, error: error) {
-                let contentFilePath = cacheDirectoryPath.stringByAppendingPathComponent("content.json")
-                let string: NSString = readabilityResult.encode()
-                return string.writeToFile(contentFilePath, atomically: true, encoding: NSUTF8StringEncoding, error: error)
-            }
+    func put(url: NSURL, _ readabilityResult: ReadabilityResult) throws {
+        let error: NSError! = NSError(domain: "Migrator", code: 0, userInfo: nil)
```

Comment:
We can clean this up a bit and move the NSError instantiation inside the guard statement below since we don't use this error anywhere else in this method.